### PR TITLE
add a plugin to measure node state

### PR DESCRIPTION
**Pull request by Kennric**
fixes issue #279

## Changes in this PR.

- [X] Add NodeActive plugin

## Testing this PR.

1. docker-compose up test
2. NodeActive tests are run and pass
3. deploy to staging
4. observer active state measurements being stored

### Expected Output.

```
$ rake spec
[...]
```

@osuosl/devs

### CODE REVIEW

**Comment by Kennric** [comment]
When this is merged, the code in #277 can be simplified, since we won't need to infer activity from other measurements

File: plugins/NodeActive/plugin.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+require 'logging'
+require 'sequel'
+require_relative '../../lib/BasePlugin/plugin.rb'
+require_relative '../../environment.rb'
+require_relative '../../models.rb'
+require_relative '../../logging/logs'
+
+# Disk Sizes data plugin
```
Comment (hsolorza) [inline plugins/NodeActive/plugin.rb]: Should be Node Active

File: plugins/NodeActive/plugin.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      MyLog.log.warn "NodeActive: No node state information for #{fqdn}"
+      raise "NodeActive: No active information for #{fqdn}"
+    end
+
+    # check if node resource exist, otherwise set it to default
+    node_resource = @database[:node_resources].where(name: fqdn).get(:id)
+    node_resource = default_node(fqdn, node_info['type'],
+                                 node_info['cluster']) unless node_resource
+
+    # Insert data into disk_size_measurements table
```
Comment (hsolorza) [inline plugins/NodeActive/plugin.rb]: Should be node_active_measurements

**Comment by alxngyn** [comment]
++

**Pull request merged**